src/commands/archive.ts

```diff
@@ -20,6 +25,18 @@ import isRoot from '../util/isRoot'
 
 let undoArchiveTimer: number
 
+/** Archives a note path and its optional target path if =note/=path structure exists. */
+const archiveNoteAndTargetThought =
```

Comment:
No need for this to be a standalone function since it is so small and is only used in one place.

Factoring it out into a function actually gets in the way of eliminating some duplication. Once you inline it, look for opportunities to simplify the conditional logic.